### PR TITLE
chat command handling

### PR DESCRIPTION
**Pull request by Trompettesib**

### CODE REVIEW

**Comment by Trompettesib** [comment]
I need to comment what I did btw

File: cubic-server/command_parser/commands/QuestionMark.hpp [hunk 1]
```diff
@@ -5,11 +5,11 @@
 
 namespace command_parser {
 struct QuestionMark : public CommandBase {
-    QuestionMark() : CommandBase("?", "/help [<command>] (question mark)") {}
+    QuestionMark() : CommandBase("?", "/help [<command>] (question mark)", 0) {}
```
Comment (STMiki) [inline cubic-server/command_parser/commands/QuestionMark.hpp:8]: /? instead of /help

File: cubic-server/protocol/ServerPackets.hpp [hunk 2]
```diff
@@ -26,6 +26,7 @@ namespace protocol
         QueryBlockEntityTag = 0x01,
         ChangeDifficulty = 0x02,
         ChatMessage = 0x03,
+        ChatCommand = 0x04,
```
Comment (STMiki) [inline cubic-server/protocol/ServerPackets.hpp:29]: 0x03, we are still in 1.19 (I think)
And yes, ChatMessage should be 0x04, but that does not fix the chat...

**Review by STMiki — COMMENTED** [review]
Why did you implement the permissions level system ? didn't we discuss that it will be reworked ? With like real permissions and stuff

File: cubic-server/protocol/Structures.hpp [hunk 3]
```diff
@@ -50,5 +50,11 @@ namespace protocol
         int32_t type;
         nbt::Compound data;
     };
+
+    struct argumentSignature
```
Comment (huntears) [inline cubic-server/protocol/Structures.hpp:54]: That should be in PascalCase

**Review by huntears — CHANGES_REQUESTED** [review]

**Review by huntears — APPROVED** [review]

**Review by STMiki — APPROVED** [review]

**Pull request merged**